src/tsdb.c

```diff
@@ -580,6 +593,119 @@ int SeriesAddSample(Series *series, api_timestamp_t timestamp, double value) {
     return TSDB_OK;
 }
 
+void CompactionDelRange(Series *series, timestamp_t start_ts, timestamp_t end_ts) {
+    CompactionRule *rule = series->rules;
+    if (!rule)
+        return;
+    RedisModuleCtx *ctx = RedisModule_GetThreadSafeContext(NULL);
+
+    while (rule) {
+        const timestamp_t ruleTimebucket = rule->timeBucket;
```

Comment:
@OfirMos I think we are missing a case where we are deleting several buckets.
I suggest to calculate all buckets in the start->end range, compare it to the downsampled key in the same time range and override with the calculate (that would also mean deleting samples if needed)

Reply:
@danni-m We aren't this is the part which deletes whole buckets:

            // ---- handle continuous deletion ----
__continous_del:
            if (rule_deletion_end >= rule_deletion_start) {
                RedisModuleKey *key;
                Series *destSeries;
                if (!GetSeries(ctx, rule->destKey, &key, &destSeries, REDISMODULE_READ)) {
                    RedisModule_Log(ctx, "verbose", "%s", "Failed to retrieve downsample series");
                    continue;
                }

                SeriesDelRange(destSeries, rule_deletion_start, rule_deletion_end);
            }
        }